compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -328,28 +334,32 @@ impl<'tcx> TyCtxt<'tcx> {
         where
             F: FnMut(ty::Region<'tcx>) -> bool,
         {
-            fn visit_binder<T: TypeFoldable<'tcx>>(&mut self, t: &Binder<T>) -> bool {
+            fn visit_binder<T: TypeFoldable<'tcx>>(
+                &mut self,
+                t: &Binder<T>,
+            ) -> ControlFlow<(), ()> {
                 self.outer_index.shift_in(1);
                 let result = t.as_ref().skip_binder().visit_with(self);
                 self.outer_index.shift_out(1);
                 result
             }
 
-            fn visit_region(&mut self, r: ty::Region<'tcx>) -> bool {
-                match *r {
+            fn visit_region(&mut self, r: ty::Region<'tcx>) -> ControlFlow<(), ()> {
+                let should_break = match *r {
```

Comment:
I think it's better to also change `self.callback` to return `ControlFlow` here.

Reply:
This is for `any_free_region_meets`, I think it makes more sense to require `bool` so it can be used like `Iterator::any`.

Reply:
hmm, makes sense :thinking: 

Can you instead move the `if should_break` into the match arm and directly return `ControlFlow::CONTINUE` in the `ReLateBound` case?